time-source/src/main/java/org/springframework/cloud/stream/module/time/TimeSource.java

```diff
@@ -18,37 +18,30 @@
 
 import java.text.SimpleDateFormat;
 import java.util.Date;
-import java.util.concurrent.TimeUnit;
 
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.context.properties.EnableConfigurationProperties;
 import org.springframework.cloud.stream.annotation.EnableModule;
 import org.springframework.cloud.stream.annotation.Source;
-import org.springframework.context.annotation.Bean;
+import org.springframework.cloud.stream.module.PeriodicTriggerConfiguration;
+import org.springframework.context.annotation.Import;
 import org.springframework.integration.annotation.InboundChannelAdapter;
 import org.springframework.integration.annotation.Poller;
-import org.springframework.scheduling.Trigger;
-import org.springframework.scheduling.support.PeriodicTrigger;
 
 /**
  * @author Dave Syer
  * @author Glenn Renfro
  */
 @EnableModule(Source.class)
+@Import(PeriodicTriggerConfiguration.class)
 @EnableConfigurationProperties(TimeSourceProperties.class)
 public class TimeSource {
 
 	@Autowired
 	private TimeSourceProperties properties;
 
-	@Bean
-	public Trigger trigger() {
-		PeriodicTrigger trigger = new PeriodicTrigger(properties.getFixedDelay(), TimeUnit.valueOf(properties.getTimeUnit()));
-		trigger.setInitialDelay(properties.getInitialDelay());
-		return trigger;
-	}
-
-	@InboundChannelAdapter(value = Source.OUTPUT, poller = @Poller(trigger = "trigger", maxMessagesPerPoll = "1"))
+	@InboundChannelAdapter(value = Source.OUTPUT, poller = @Poller(trigger = PeriodicTriggerConfiguration
```

Comment:
Just refactoring to use /validate common configuration. It seems like more work to back out and issue another PR to do this but if you feel strongly about it...